### ISSUE TITLE
Workspace Delete failing on empty workspaces

### ISSUE THREAD

**Issue by FGS-CThomas** [hashicorp/terraform#22402]
### Terraform Version

```
Terraform v0.12.6
+ provider.aws v2.12.0
+ provider.null v2.1.2
+ provider.random v2.1.2
+ provider.template v2.1.2
```

### Debug Output

[This](https://gist.github.com/FGS-CThomas/3063eba21eb37dd2be28f7f7ec13f6df) is the debug output from a `terraform state list` command inside of a supposedly empty workspace.  The workspace itself consists of outputs and a bunch of resources with no instances.  

EG: 
```
"resources": [
    {
      "module": "module.websites.module.tls",
      "mode": "data",
      "type": "aws_acm_certificate",
      "name": "reused",
      "each": "list",
      "provider": "provider.aws",
      "instances": []
    },
```
If I switch out of the aforementioned workspace and attempt to delete it I get[ this debug output](https://gist.github.com/FGS-CThomas/64cd4607c031d237b749f69754f1ee12).

### Expected Behavior
When attempting to delete an empty workspace I expect the workspace to be deleted.  Right?

### Actual Behavior
Instead, I get the following output...
```
Workspace "test-workspace" is not empty.

Deleting "test-workspace" can result in dangling resources: resources that
exist but are no longer manageable by Terraform. Please destroy
these resources first.  If you want to delete this workspace
anyway and risk dangling resources, use the '-force' flag.
```

### Steps to Reproduce
1. terraform init
2. terraform workspace new zombie
3. terraform apply
4. terraform plan -destroy -out destroy.tfplan
5. terraform apply destroy.tfplan
6. terraform workspace select default
7. terraform workspace delete zombie

### Additional Context
I note that if I *manually* delete the `resources` array from the JSON state file that the workspace can be deleted.  This is obviously not a good solution but it suggests to me that Terraform is failing to clean up `resources` objects once they no longer have any instances?

**Comment by davegallant** [comment]
I experienced this when planning a destroy, and then applying the plan. By not using the `destroy` command, `PruneResourceHusks()` is not run, and leaves anything with a `count` key still in the resources list in the state file.

See: https://github.com/hashicorp/terraform/blob/7e29b9b5d4d17a94ba1e00fdaabd7a1f4828d5d4/terraform/context.go#L451

**Comment by Rinzler081** [comment]
I’m running into the same issue with terraform v12.26 and aws provider 2.70. Any ideas?
I only can get around deleting the workspace with a force flag

**Comment by apparentlymart** [comment]
Hi all!

It looks like we later got a duplicate of this in #23269 but didn't notice that there was already this issue open about the problem. That issue was subsequently fixed in #29754, and so modern Terraform uses a more precise test for whether a state is actually tracking remote objects, and will also return a more helpful error message if there _are_ any such objects, specifying their addresses so that it's easier to respond by looking them up to see what they represent and whether they are important to preserve.

Thanks for reporting this, and sorry we missed it when fixing it later!
